contracts/FIFOMarket.sol

```diff
@@ -125,7 +125,7 @@ contract FIFOMarket is
     address[] memory suppliers = new address[](_queueLength());
     for (uint256 i = _queueHeadIndex; i < _queueNextInsertIndex; i++) {
       uint256 removalAmount = _removal.balanceOf(address(this), _queue[i]);
-      address supplier = _removal.vintage(_queue[i]).supplier;
+      address supplier = _removal.supplierAddressFromTokenId(_queue[i]);
```

Comment:
Since the supplier address is encoded in the removal token id itself, we don't have to perform a struct lookup to know who to pay anymore.  hopefully gas savings... wondering if there's any gas still to be saved by defining this method here in the `FIFOMarket` contract in order to avoid the external call to the removal contract... since it's just calling a view function, maybe the gas is exactly the same?  obviously better to avoid implementing the same method here if we can help it so we're not maintaining this logic in two places.

Reply:
I think you could drop the `supplierAddressFromTokenId` and friends into a library file like we did with `ScheduleUtils.sol` then both contracts can inline it.  Seems like an optimization we can experiment with later though.

Reply:
did it! makes a ton of sense in this case, great opportunity to learn about solidity libraries